src/Events/BaseEvent.php

```diff
@@ -16,6 +16,28 @@
  */
 abstract class BaseEvent extends Event implements Serializable
 {
+    public function __serialize(): array
```

Comment:
Maybe
```
    public function __serialize(): array
    {
        return [
            'base_event_class' => get_class($this),
            'subject' => $this->getSubject(),
            'marking' => $this->getMarking(),
            'transition' => $this->getTransition(),
            'workflow' => [
                'name' => $this->getWorkflowName(),
            ],
        ];
    }

    public function __unserialize(array $data): void
    {
        $this->subject = $data['subject'];
        $this->marking = $data['marking'];
        $this->transition = $data['transition'] ?? null;
        $workflowName = $data['workflow']['name'] ?? null;
        $this->workflow = Workflow::get($this->subject, $workflowName);
    }

    public function serialize()
    {
        return serialize($this->__serialize());
    }

    public function unserialize($serialized)
    {
        $this->__unserialize(unserialize($serialized));
    }
```

Reply:
I've updated the code, updated as suggested, works good.